handler.go

```diff
@@ -233,6 +240,45 @@ func mountStaticFiles(logger *slog.Logger, mux *http.ServeMux) error {
 	})
 }
 
+// Requires basic auth username/password authentication on all endpoints
+// https://en.wikipedia.org/wiki/Basic_access_authentication
+type basicAuthMiddleware struct {
+	username string
+	password string
+}
+
+func (m basicAuthMiddleware) Middleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		basicAuthHeader := r.Header.Get("Authorization")
+		if basicAuthHeader == "" {
+			w.Header().Set("WWW-Authenticate", `Basic realm="riverui restricted"`)
+			w.WriteHeader(http.StatusUnauthorized)
+			return
+		}
+
+		if !strings.HasPrefix(basicAuthHeader, "Basic ") {
+			w.Header().Set("WWW-Authenticate", `Basic realm="riverui restricted"`)
+			w.WriteHeader(http.StatusUnauthorized)
+			return
+		}
+
+		data, err := base64.StdEncoding.DecodeString(basicAuthHeader[len("Basic "):])
```

Comment:
Do you know about [`Request.BasicAuth`](https://pkg.go.dev/net/http?utm_source=godoc#Request.BasicAuth)? Probably better to use that than implement it yourself.